src/err/mod.rs

```diff
@@ -192,11 +207,11 @@ impl PyErr {
     /// If `ty` does not inherit from `BaseException`, then a `TypeError` will be returned.
     ///
     /// If calling `ty` with `args` raises an exception, that exception will be returned.
-    pub fn from_type<A>(ty: &PyType, args: A) -> PyErr
+    pub fn from_type_bound<A>(ty: Bound<'_, PyType>, args: A) -> PyErr
```

Comment:
Arguably this could also just take a `Py<PyType>`, but then we would need a different name I guess...

Reply:
Ugh, yes. How about `new_with_type`? 

I'm also kinda ok with leaving this as-is; it's a bit easier to get a `Bound<PyType>` than a `Py<PyType>` (e.g. with `PyAnyMethods::get_type()`, or just `.downcast_into()`)

Reply:
What do you prefer? I think its fine either way.

Reply:
For simplicity of migration let's just keep this as is, given there's no obvious answer.